MatchServer/MMatchServer.cpp

```diff
@@ -562,6 +568,7 @@ void MMatchServer::Destroy()
 	m_AsyncProxy.Destroy();
 	MGetMatchShop()->Destroy();
 	m_SafeUDP.Destroy();
+	MGetGunGame()->~GunGame();
```

Comment:
The static object in `GunGame::GetInstance` is automatically destroyed at the end of the program if it is initialized, so this causes it to be destroyed twice, which is undefined behavior. If you really need to do this (it doesn't seem that you do), you need to make a normal Destroy function instead.